### PR TITLE
added linuxmint-19-xfce 32bit box

### LINKED ISSUE #68: Custom vagrant box
- [x] Build a custom vagrant box based on LinuxMint 19 Xfce. This will speed up testing and usage (when users build a new ltsp server).
   - https://www.linuxmint.com/download.php
   - https://www.vagrantup.com/docs/virtualbox/boxes.html
   - https://www.engineyard.com/blog/building-a-vagrant-box
   - https://www.skoblenick.com/vagrant/creating-a-custom-box-from-scratch/
   - http://blog.ruilopes.com/from-iso-to-vagrant-box.html
   - http://jayunit100.blogspot.com/2014/10/building-vagrant-virtualbox-box-from-iso.html

 - [x] Add a configuration variable on `settings.sh` for the vagrant base box, so that users can choose another base system (instead of LinuxMint Xfce). As long as it is an Ubuntu derivative all the provision scripts should work fine.

- [x] Write a blog or wiki page about how to build a custom vagrant box for the ltsp-server, so that the users can use another flavor of Ubuntu for their ltsp server.

### PR DESCRIPTION
**Pull request by d78ui98**
closes #68

### CODE REVIEW

**Comment by d78ui98** [comment]
Maybe we should have a different branch named `mint`. Here we can have all 6 linux mint boxes.

Also Note that dhcp server is still ubuntu `bionic`. If I change it to `linuxmint-19-xfce 32bit` it would be too heavy for `512mb` ram. Let me know what should I do here.

File: install.sh [hunk 1]
```diff
@@ -24,13 +21,9 @@ echo 'IPAPPEND=3' >> /etc/ltsp/update-kernels.conf
 /usr/share/ltsp/update-kernels
 
 # Configure dnsmasq
+echo "port=5353" >> /etc/dnsmasq.conf
```
Comment (dashohoxha) [inline install.sh:24]: Why is this needed?

**Review by dashohoxha — APPROVED** [review]
It seems OK.

**Comment by dashohoxha** [comment]
> Maybe we should have a different branch named `mint`. Here we can have all 6 linux mint boxes.

No, this is not right. LinuxMint19 is based on Ubuntu 18.04 LTSP, so it is on the `bionic` category too. 

> Also Note that dhcp server is still ubuntu bionic. If I change it to linuxmint-19-xfce 32bit it would be too heavy for 512mb ram. Let me know what should I do here.

You should do nothing here. I am sorry that you ask this question. The `dhcp` server is just for testing, so it doesn't matter what it is. For example if you can build it on Arch Linux it is still OK. Also you can use the same dhcp server (based on `bionic`) for the `buster` branch too, and this is OK.

**Comment by d78ui98** [comment]
Okay, thats Awsome.

File: install.sh (hunk 1 above)
Comment (d78ui98) [reply install.sh:24]: See the issue is in `tara` dnsmasq is conflicting with systemd-resolved. It gives this error
```
dnsmasq[4285]: failed to create listening socket for port 53: address already in use
```
As a result dnsmasq service is unable to start. So a good option was to switch it to another port.

Comment (dashohoxha) [reply install.sh:24]: I am not sure whether this is a good solution. But if everything works it is fine. I am afraid that it may break something.

Another solution is discussed here (and again I am not sure whether this is good or not):
https://askubuntu.com/questions/1032450/how-to-add-dnsmasq-and-keep-systemd-resolved-18-04

This also has a discussion about a possible problem with dnsmasq: http://dashohoxha.fs.al/ltsp-pnp/#sec-3-4
I am not sure whether it is the same problem or something else.

Comment (d78ui98) [reply install.sh:24]: Yes. I saw that one as well. At some point it involves turning on and off the wired connection with `nmcli`. Doing that just freezes vagrant box. 
It might work for actual  or virtual machine. But not for vagrant box
